File: MIDIJava/src/io/github/seanboyy/Midi/MidiEvent.java
package io.github.seanboyy.Midi;

import java.util.ArrayList;

public class MidiEvent implements Event {
    private VariableLengthValue timeDelta;
    private byte eventType;
    private ArrayList<Byte> eventData;

    MidiEvent(VariableLengthValue timeDelta, byte eventType, ArrayList<Byte> eventData) {
        this.timeDelta = new VariableLengthValue(timeDelta);
        this.eventType = eventType;
        this.eventData = new ArrayList<>();
        this.eventData.addAll(eventData);
    }

    @Override
    public byte[] toBitString() {
        byte[] bitString = new byte[getLength()];
        int[] cursor = {0};
        bitString = StreamingHelper.streamObjectToBitString(cursor, bitString, timeDelta, timeDelta.getLength());
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, eventType);
        for (Byte b : eventData) {
            bitString = StreamingHelper.streamValueToBitString(cursor, bitString, b);
        }
        return bitString;
    }

    @Override
    public int getLength() {
        return timeDelta.getLength() + 1 + eventData.size();
    }
}

File: MIDIC#/MIDILibrary/MIDILibrary/MidiEvent.cs
using System.Collections.Generic;

namespace MIDILibrary
{
    public class MidiEvent : IEvent
    {
        public VariableLengthValue TimeDelta { get; }
        public byte EventType { get; }
        public List<byte> EventData { get; }

        public MidiEvent(VariableLengthValue timeDelta, byte eventType, IEnumerable<byte> eventData)
        {
            TimeDelta = new VariableLengthValue(timeDelta);
            EventType = eventType;
            EventData = new List<byte>();
            EventData.AddRange(eventData);
        }

        public uint GetLength()
        {
            return TimeDelta.Length + 1 + (uint)EventData.Count;
        }

        public byte[] ToBitString()
        {
            var bitString = new byte[GetLength()];
            uint cursor = 0;
            StreamingHelper.StreamObjectToBitString(ref cursor, ref bitString, TimeDelta, TimeDelta.Length);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, EventType);
            foreach (var b in EventData)
            {
                StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, b);
            }
            return bitString;
        }
    }
}
